src/palimpzest/solver/query_strategies.py

```diff
@@ -0,0 +1,258 @@
+from palimpzest.constants import PromptStrategy
+from palimpzest.elements import DataRecord
+from palimpzest.generators import DSPyGenerator, ImageTextGenerator
+# from palimpzest.profiler import Stats
+from palimpzest.solver.task_descriptors import TaskDescriptor
+
+from typing import Any, Dict, List, Tuple
+
+import base64
+import json
+
+
+def _construct_query_prompt(td: TaskDescriptor, doc_type: str, generate_field_names: List[str]) -> str:
```

Comment:
This is better than what we used to have, but seems like we still need an intermediate representation of data operations so we can combine novel optimizations in a more principled way than we do now.

This is worth spending some time brainstorming and discussing.

Reply:
Yeah, I don't think this is finished work, I was having a hard time deciding what the interface between `Solver`, `Generator`, and the functions in this file should be. I think this is worthy of a small meeting early next week to brainstorm.